crates/wash-cli/src/app/mod.rs

```diff
@@ -268,6 +284,36 @@ async fn put_model(cmd: PutCommand) -> Result<PutModelResponse> {
     }
 }
 
+/// Retrieve the the model name (possibly missing) of a given [`AppManifest`]
+pub(crate) fn get_model_name(manifest: &AppManifest) -> Result<&str> {
+    match manifest {
+        AppManifest::ModelName(model_name) => Ok(model_name),
+        AppManifest::SerializedModel(manifest) => {
+            if let Some(metadata) = manifest.get("metadata") {
+                if let Some(name) = metadata.get("name") {
+                    return name
+                        .as_str()
+                        .context("metadata.name in provided manifest is not a string");
+                }
+            }
+            bail!("unexpected format for manifest, did not contain metadata.name");
+        }
+    }
+}
+
+/// Retrieve the version of a given [`AppManifest`], returning None if the manifest
+/// does not contain a version (or is not the type to contain a version)
+pub(crate) fn get_model_version(manifest: &AppManifest) -> Option<&str> {
+    match manifest {
+        AppManifest::ModelName(_) => None,
+        AppManifest::SerializedModel(manifest) => manifest
```

Comment:
Nit: I don't know if this returns the wadm `Manifest` type, but if it does, there is a [`version`](https://docs.rs/wadm-types/latest/wadm_types/struct.Manifest.html#method.version) method

Reply:
It does not! That's a plain YAML object (`serde_yaml::Value`) in there, unfortunately.